src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -100,12 +82,34 @@ private ArgumentCaptor(Class<? extends T> clazz) {
      * @return null or default values
      */
     public T capture() {
-        Mockito.argThat(capturingMatcher);
+        createCustomMatcher(capturingMatcher);
         return handyReturnValues.returnFor(clazz);
     }
 
     /**
+     * Use it to capture the argument that satisfies the matcher. This method <b>must be used inside of verification</b>.
+     * <p>
+     * Internally, this method registers a special implementation of an {@link ArgumentMatcher}.
+     * This argument matcher stores the argument value that satisfies the matcher so that you can use it later to perform assertions.  
+     * <p>
+     * See examples in javadoc for {@link ArgumentCaptor} class.
+     * 
+     * @return null or default values
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+	public T captureIf(final ArgumentMatcher matcher) {
+    	createCustomMatcher(customCapturingMatcher = new CustomCapturingMatcher<T>(matcher));
+    	return handyReturnValues.returnFor(clazz);
+    }
+
+    public T createCustomMatcher(final ArgumentMatcher<T> matcher) {
```

Comment:
It should be private. By mistake is public ( if you see the method implementation, you will see that somehow the used matcher is registered. Because we have an CapturingMatcher and a CustomCapturingMatcher, I considered that it's better to use method if something changes in the future.)